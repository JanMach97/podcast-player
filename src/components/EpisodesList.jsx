import React from "react";
import map from "lodash.map";
import { Link } from "react-router-dom";

const EpisodesList = props => {
  return (
    <div className="episodes-list">
      {map(props.episodes, (episode, i) => {
        return (
          <div className="episode" key={i}>
            <div className="thumbnail">
              <i
                onClick={() => props.selectEpisode(episode, i)}
                className={
                  props.playing && props.index === i
                    ? "far fa-pause-circle"
                    : "far fa-play-circle"
                }
              />
              <img src={episode.image_url} alt="episode" />
            </div>
            <Link to={`/episodes/${episode.episode_id}`}>
              <h3>{episode.title}</h3>
            </Link>
            <p>Show Duration: {formatTime(episode.duration / 1000)}</p>
          </div>
        );
      })}
    </div>
  );
};

const formatTime = seconds => {
  const sec = parseInt(seconds % 60);
  const min = parseInt((seconds / 60) % 60);
  return `${min}:${sec < 10 ? "0" + sec : sec}`;
};

export default EpisodesList;
