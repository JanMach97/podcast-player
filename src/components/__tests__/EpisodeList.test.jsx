import * as React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import EpisodesList from '../EpisodesList'

configure({ adapter: new Adapter() });

describe('EpisodesList', () => {
  it('renders without crashing', () => {
    shallow(<EpisodesList />)
  })
})