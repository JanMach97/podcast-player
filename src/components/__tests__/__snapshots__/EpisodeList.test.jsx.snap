// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodesList renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EpisodesList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "episodes-list",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "episodes-list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`EpisodesList renders without crashing with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EpisodesList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "episodes-list",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "episodes-list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
